sdk/common/src/main/java/io/opentelemetry/sdk/internal/ComponentId.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.internal;
+
+import io.opentelemetry.api.common.AttributesBuilder;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.annotation.Nullable;
+
+/**
+ * The component id used for SDK health metrics. This corresponds to the otel.component.name and
+ * otel.component.id semconv attributes.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public abstract class ComponentId {
+
+  /**
+   * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+   * any time.
+   */
+  public interface StandardType {
```

Comment:
Since the only implementation of this is the StandardExporterType enum, do we need an interface or can we just stick this method on StandardExporterType?

Reply:
I think I get it:

- You [say](https://github.com/open-telemetry/opentelemetry-java/pull/7265#discussion_r2073435454) that these utilities can be used outside of opentelemetry-java (at your own risk)
- To do so, you need to call `new SemconvExporterMetrics`, which accepts a ComponentId. 
- ComponentId has a package private constructor, so the only way you can actually obtain one is via `generateLazy(StandardType)`

But there's also `generateLazy(String)`, so I don't actually think we need a StandardType interface.

Reply:
I added this because of the upcoming additional non-exporter standard types, e.g. the span processors or the periodic metric reader, which don't really fit into the same category (those don't need the `signal` attribute).

However, let's keep it simple in this PR, so I'll remove it and we can decide in the followup PRs for the processors whether having an interface makes sense or not.